packages/modules/vehicles/common/vehicle_utils.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python3
+
+# collection of common utilities
+# used by vehicle soc/range modules
+
+from typing import Union, Optional
+import logging
+from datetime import datetime
+from time import time
+from jwt import decode
+from helpermodules.pub import Pub
+from json import dump, dumps, load, loads
+import paho.mqtt.subscribe as subscribe
+import paho.mqtt.publish as publish
+import paho.mqtt.client as mqtt
+from modules.common.configurable_vehicle import ConfigurableVehicle
+from control import data
+from modules.common.component_state import CarState
+
+
+DEF_LOGFILTER = '__'
+
+log = logging.getLogger(__name__)
+
+
+# logUtils - filtered logging
+# if logFilter is defined log filtered messages only otherwise log normally on INFO level
+class logUtils:
+
+    def __init__(self, filter: Optional[str] = None):
+        if filter:
+            self.logFilter = filter
+        else:
+            self.logFilter = DEF_LOGFILTER
+
+    def set_logFilter(self, filter: str = ''):
+        self.logFilter = filter
+
+    # get logFilter via mqtt
+    def get_logFilter(self, vnum: str) -> str:
+        mq = mqttUtils(self)
+        try:
+            msg = mq.mqtt_get_topic("openWB/vehicle/" + vnum + "/soc_module/config", "localhost")
+            config = loads(str(msg.payload, 'utf-8'))
+            if 'logFilter' in config['configuration']:
+                self.logFilter = config['configuration']['logFilter']
+                if self.logFilter is None:
+                    self.logFilter = DEF_LOGFILTER
+            else:
+                self.logFilter = DEF_LOGFILTER
+        except Exception as e:
+            log.exception("get_logFilter mqtt error: %s", e)
+            self.logFilter = DEF_LOGFILTER
+        self.infoLog('Q', 'get_logFilter: logFilter=' + self.logFilter)
+        return self.logFilter
+
+    def infoLog(self, filter: str, txt: str):
+        if self.logFilter != DEF_LOGFILTER:
+            if filter in self.logFilter or 'A' in self.logFilter:
+                log.info('(' + filter + '): ' + txt)
+        else:
+            log.info(txt + ', filter=' + filter + ', logFilter=' + self.logFilter)
```

Comment:
Der Benutzer kann zwischen den Loglevels Details, Info und Warnung/Error wählen. Warum sollte das Modul diese Entscheidung treffen?

Reply:
Die Loglevel Details, Info, Warn/Error sind ziemlich grob.
Im Fall der Fälle würde zu viel unwichtiges gelogged werden.
Das smarteq Modul unterscheidet diese Teilbereiche:
    # A: ALL
    # f: fetch_soc
    # i: initialization
    # m: websocket message
    # M: websocket message details
    # q: mqtt logging
    # Q: mqtt logging details
    # r: smart eq rest api
    # s: state changes
    # t: token refresh
    # T: token refresh details
    # u: authentication
    # U: authentication details
    # w: webSocket interface
    # W: webSocket interface details

je nach Problem/Bedarf sollen nur diese Logs erzeugt werden.

Reply:
Das würde ich doch gerne so lassen. 
Die Log-Filter können vom Benutzer in der Modul-Konfiguration geändert werden.
Die Logging-Level wären zu ungenau und würden je nach Problem zuviel unnötige Logs erzeugen.